polaris/evaluate/_metric.py

```diff
@@ -84,55 +164,241 @@ class Metric(Enum):
+        return self
+
     @property
     def fn(self) -> Callable:
         """The callable that actually computes the metric"""
-        return self.value.fn
+        return self.metric.value.fn
 
     @property
     def is_multitask(self) -> bool:
         """Whether the metric expects a single set of predictions or a dict of predictions."""
-        return self.value.is_multitask
+        return self.metric.value.is_multitask
 
     @property
     def y_type(self) -> bool:
-        """Whether the metric expects preditive probablities."""
-        return self.value.y_type
+        """The type of input the metric expects."""
+        return self.metric.value.y_type
 
     def score(
-        self, y_true: np.ndarray, y_pred: np.ndarray | None = None, y_prob: np.ndarray | None = None
+        self,
+        y_true: GroundTruth,
+        y_pred: BenchmarkPredictions | None = None,
+        y_prob: BenchmarkPredictions | None = None,
     ) -> float:
-        """Endpoint for computing the metric.
+        """Compute the metric.
 
-        For convenience, calling a `Metric` will result in this method being called.
+        Args:
+            y_true: The true target values.
+            y_pred: The predicted target values, if any.
+            y_prob: The predicted target probabilities, if any.
+        """
+        pred = prepare_predictions(y_pred, y_prob, self.y_type)
+        kwargs = prepare_metric_kwargs(
+            y_true.as_array("y"),
+            pred.flatten(),
+            self.y_type,
+        )
+        return self.fn(**kwargs, **self.metric.value.kwargs)
+
+    def __call__(
+        self,
+        y_true: GroundTruth,
+        y_pred: BenchmarkPredictions | None = None,
+        y_prob: BenchmarkPredictions | None = None,
+    ) -> float:
+        """
+        For convenience, calling a `Metric` is the same as calling its `score` method.
 
         ```python
-        metric = Metric.mean_absolute_error
+        metric = Metric(metric="mean_absolute_error")
         assert metric.score(y_true=first, y_pred=second) == metric(y_true=first, y_pred=second)
         ```
         """
-        if y_pred is None and y_prob is None:
-            raise ValueError("Neither `y_pred` nor `y_prob` is specified.")
+        return self.score(y_true, y_pred, y_prob)
 
-        if self.y_type == "y_pred":
-            if y_pred is None:
-                raise ValueError(f"{self} requires `y_pred` input. ")
-            pred = y_pred
-        else:
-            if y_prob is None:
-                raise ValueError(f"{self} requires `y_prob` input. ")
-            pred = y_prob
+    # For any given benchmark, all of its metrics need to have a unique name.
+    # Within the scope of a given benchmark metric is thus uniquely identified by its name.
 
-        kwargs = {"y_true": y_true, self.y_type: pred}
-        return self.fn(**kwargs, **self.value.kwargs)
+    def __hash__(self) -> int:
+        return hash(self.name)
```

Comment:
**Question**
If the content of `config` matters to the behavior of the metric, then this is probably not accurate.

Reply:
My thinking here was to shift the responsibility of defining a unique string ID per metric to the user. This was we could also show this unique ID in the UI for the Hub. Thinking about this again though, that actually seems like unnecessary friction and a worse UX.